plugins/design-system/src/App.tsx

```diff
@@ -272,12 +285,22 @@ export function App() {
                                     }}
                                 />
                                 <div className="color-label">{color.title}</div>
-                                <button className="copy-button" onClick={() => handleAddColors([color])}>
+                                <button
+                                    className="copy-button"
+                                    disabled={!isAllowedToCreateColorStyle}
+                                    title={isAllowedToCreateColorStyle ? undefined : "Insufficient permissions"}
+                                    onClick={() => handleAddColors([color])}
+                                >
                                     Add
                                 </button>
                             </div>
                         ))}
-                        <button className="action-button" onClick={() => handleAddColors(filteredColorItems)}>
+                        <button
+                            className="action-button"
+                            disabled={!isAllowedToUpsertColor}
+                            title={isAllowedToUpsertColor ? undefined : "Insufficient permissions"}
```

Comment:
I guess this could be a helper, since it's done more than once here, something like this:

```ts
const getPermissionTitle = (isAllowed) => isAllowed ? undefined : "Insufficient permissions"
```